src/DataServices/Core/MemberDataService.php

```diff
@@ -139,19 +138,34 @@ public function getPage(RequestParameters $requestParameters, array $queryParams
                 }
             }
 
-            $order = $requestParameters->getParameters()['order'];
-            $orderBy = $requestParameters->getParameters()['orderBy'];
-            $pageSize = $requestParameters->getParameters()['pageSize'];
-            $firstResult = $pageSize * $requestParameters->getParameters()['pageNumber'];
-
             if (!empty($whereStatement)) {
                 $this->queryBuilder
                     ->where($whereStatement)
                     ->setParameters($whereParameters);
             }
 
+            $order = $requestParameters->getParameters()['order'];
+            $orderBy = $requestParameters->getParameters()['orderBy'];
+            $pageSize = $requestParameters->getParameters()['pageSize'];
+            $firstResult = $pageSize * $requestParameters->getParameters()['pageNumber'];
+
             if (isset($orderBy)) {
-                $this->queryBuilder->orderBy($orderBy, $order);
+                switch ($orderBy) {
+                    case 'lastName' :
+                    case 'firstName' :
+                    case 'email' :
+                        $this->queryBuilder->orderBy("m.$orderBy", $order);
+                        break;
+                    case 'username':
+                        $this->queryBuilder->orderBy("u.$orderBy", $order);
+                        break;
+                    case 'poleLabel' :
+                        $this->queryBuilder->orderBy("pole.label", $order);
+                        break;
+                    case 'positionLabel' :
+                        $this->queryBuilder->orderBy("p.label", $order);
+                        break;
+                }
             }

```

Comment:
m, u, p are the abbreviations of member username and positionLabel in the sql query.